packages/flutter/lib/src/material/dropdown.dart

```diff
@@ -644,17 +658,23 @@ class _DropdownButtonState<T> extends State<DropdownButton<T>> with WidgetsBindi
           mainAxisAlignment: MainAxisAlignment.spaceBetween,
           mainAxisSize: MainAxisSize.min,
           children: <Widget>[
-            // If value is null (then _selectedIndex is null) then we display
+            // If value is null (then _selectedIndex is null) or if disabled then we display
             // the hint or nothing at all.
             new IndexedStack(
-              index: _selectedIndex ?? hintIndex,
+              index: _enabled ? (_selectedIndex ?? hintIndex) : hintIndex,
               alignment: AlignmentDirectional.centerStart,
               children: items,
             ),
             new Icon(Icons.arrow_drop_down,
               size: widget.iconSize,
               // These colors are not defined in the Material Design spec.
-              color: Theme.of(context).brightness == Brightness.light ? Colors.grey.shade700 : Colors.white70
+              color: _enabled
+                  ? Theme.of(context).brightness == Brightness.light
+                  ? Colors.grey.shade700
+                  : Colors.white70
+                  : Theme.of(context).brightness == Brightness.light
+                  ? Colors.grey.shade400
+                  : Colors.white10,
```

Comment:
consider factoring this out into a separate method that uses "if" statements, it would be easier to read

Reply:
Done